x/tokenregistry/keeper/keeper.go

```diff
@@ -165,3 +165,16 @@ func (k keeper) SetFirstLockDoublePeg(ctx sdk.Context, denom string, networkDesc
 		instrumentation.PeggyCheckpoint(ctx.Logger(), instrumentation.SetFirstLockDoublePeg, "networkDescriptor", networkDescriptor, "registry", registry)
 	}
 }
+
+// TODO get the denom temporarily, will add a map to keep the data from network+address to denom
+// after confirmed we really need this to identify the denom after receive the burn event from Ethereum
+func (k keeper) GetDenomFromContract(ctx sdk.Context, networkDescriptor oracletypes.NetworkDescriptor, contract string) (string, error) {
+	entries := k.GetRegistry(ctx)
+	for _, entry := range entries.Entries {
```

Comment:
Is it possible to have an attack vector where this list is too long to traverse efficiently?

Reply:
It is good question. currently, I am not 100% sure we should get the denom this way. After confirmed, I will create a map in the keeper to store (network, address) to denom.